### PR TITLE
Add `StringPool#get?`

### LINKED ISSUE #11217: Add `StringPool#get?` variants
Right now `StringPool` only provides `get` methods: if the passed string or bytes are in the string pool, that string is returned. Otherwise a new string is created and put in the pool.

It might be sometimes useful to check if a string is in the pool already, but without creating a new entry.

One use case I have in mind is an http router: each known segment can be put into the string pool, and when splitting a path to match against the router segments we can use the string pool for known segments, but create new strings for unknown segments. That would mean that for the most common scenario where valid paths are sent, no new strings are created for the path segments.

### PR DESCRIPTION
**Pull request by HertzDevil**
Resolves #11217.

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

File: src/string_pool.cr [hunk 1]
```diff
@@ -86,6 +86,23 @@ class StringPool
...
+  #
+  # ```
+  # require "string_pool"
+  #
+  # pool = StringPool.new
+  # bytes = "abc".to_slice
+  # pool.get?(bytes) # => nil
+  # pool.empty?      # => true
+  # pool.get(bytes)  # => "abc"
+  # pool.empty?      # => false
```
Comment (ysbaddaden) [inline src/string_pool.cr:100]: **suggestion**: the doc could have a second call to `get?` that would now return a String.

File: src/string_pool.cr [hunk 2]
```diff
@@ -119,11 +151,22 @@ class StringPool
...
+      entry = String.new(str, len)
+      @hashes[index] = hash
+      @values[index] = entry
+      entry
+    end
+  end
+
+  private def get?(hash : UInt64, str : UInt8*, len)
+    get(hash, str, len) { nil }
   end
```
Comment (ysbaddaden) [inline src/string_pool.cr:170]: **polish**: the method is private and called exactly once, it could be inlined right in `get?(ptr, len)`.

**Review by ysbaddaden — APPROVED** [review]
A simple yet very welcomed addition :+1:

**Comment by straight-shoota** [comment]
@HertzDevil WDYT about Julien's suggestions?

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**